### PR TITLE
Remove total difficulty from the best peer calculation

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR removes total difficulty as the criteria for the best sync peer search.

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [x] I have added sufficient documentation in code

## Testing

- [x] I have tested this code with the official test suite
- [x] I have tested this code manually

### Manual Testing

* Checked that syncing works between old binary <-> new binary
* Checked that syncing works between new binary <-> old binary
* Checked that syncing works between new binary <-> new binary

# Additional comments

Fixes EDGE-519

### CODE REVIEW

**Review by Kourin1996 — APPROVED** [review]
LGTM

**Review by dbrajovic — APPROVED** [review]

File: protocol/syncer.go [hunk 1]
```diff
@@ -295,7 +277,6 @@ func (s *Syncer) updatePeerStatus(peerID peer.ID, status *Status) {
 		status.Number,
 		"latest block hash",
 		status.Hash, "difficulty",
```
Comment (0xAleksaOpacic) [inline protocol/syncer.go]: Should we reove "difficulty" from debug log?

**Review by 0xAleksaOpacic — APPROVED** [review]
LGTM!

File: protocol/syncer.go (hunk 1 above)
Comment (zivkovicmilos) [reply protocol/syncer.go]: Brought it back in the latest commit that preserves backwards compatibility:

[138ea36](https://github.com/0xPolygon/polygon-edge/pull/524/commits/138ea36c43d91280270d57ca27bf7bda1fddb24a)

Comment (0xAleksaOpacic) [reply protocol/syncer.go]: Thanks, good catch!

**Pull request merged**